### PR TITLE
Nio azure resource

### PR DESCRIPTION
**Pull request by rickle-msft**
This PR is purely refactor and should change no functionality. It became clear that most of the helper methods written up to this point were generally operations on some sort of remote resource, so this PR introduces the abstraction of AzureResource to be a central, logical location for all those methods.

### CODE REVIEW

File: sdk/storage/azure-storage-blob-nio/src/test/resources/session-records/AzureFileSystemProviderSpecfilesystemprovidercopynorootdir.json [hunk 1]
```diff
@@ -1,11 +1,19 @@
 {
   "networkCallRecords" : [ {
     "Method" : "GET",
+<<<<<<< HEAD
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob-nio/src/test/resources/session-records/AzureFileSystemProviderSpecfilesystemprovidercopynorootdir.json]: Merge conflict here

File: sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureResource.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.nio;
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureResource.java]: If this class is for internal use only would it make sense to move it into `implementation`?

Comment (rickle-msft) [reply sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureResource.java]: I'm not sure. It's pretty integral to some of the functionality and seems like it's of the same kind as the other types in the package. I think I'll leave it here for now, and if it seems it doesn't make sense in the package any more, I can move it later.

**Pull request merged**